activerecord/test/cases/adapters/mysql2/optimizer_hints_test.rb

```diff
@@ -15,6 +15,14 @@ def test_optimizer_hints
       end
     end
 
+    def test_optimizer_hints_with_count_subquery
+      assert_sql(%r{\ASELECT /\*\+ NO_RANGE_OPTIMIZATION\(posts index_posts_on_author_id\) \*/}) do
+        posts = Post.optimizer_hints("NO_RANGE_OPTIMIZATION(posts index_posts_on_author_id)")
+        posts = posts.select(:id).where(author_id: [0, 1]).limit(5)
+        assert_equal 5, posts.count
+      end
+    end
```

Comment:
Wrapping this with `with_db_warnings_action(:raise)` will reveal the ignored hint.